frontend/src/pages/DirectoryView/SearchBar/index.jsx

```diff
@@ -4,18 +4,43 @@ import { Button } from "reactstrap";
 import "../styles.scss";
 
 class SearchBar extends Component {
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.handleSearch();
+  };
+
+  onChangeKeyword = e => {
+    this.props.onChangeKeyword(e.target.value);
+  };
+
+  onChangeLocation = e => {
+    this.props.onChangeLocation(e.target.value);
+  };
+
+  onChangeTag = e => {
+    this.props.onChangeTag(e.target.value);
+  };
+
   render() {
     return (
       <div className="searchbar-wrapper">
-        <form className="search">
+        <form className="search" onSubmit={this.handleSubmit}>
           <label>
-            <input id="search-general" type="text" />
+            <input
```

Comment:
Nice work here using `onChange` handlers. One thing that we want to ensure when we do this is that we're doing a fully "Controlled Component". What this means is that the input fields receive as props the value and when you change the field, you call the onChange handler (like you do here). What this does is allows us to not have two pieces of data that can somehow become out of sync (the parent's recordd of what's in this textbox which we hold in state and what's actually in the textbox). See [Controlled Components.](https://reactjs.org/docs/forms.html)